crates/executor/src/executor.rs

```diff
@@ -149,53 +133,54 @@ where
                 // does it exist inside cached contracts if it doesn't it is new bytecode that
                 // we are inserting inside `change`
                 if let Some(ref code) = account.info.code {
-                    if !code.is_empty() {
-                        match db.contracts.entry(account.info.code_hash) {
-                            Entry::Vacant(entry) => {
-                                entry.insert(code.clone());
-                                new_bytecodes.insert(H256(account.info.code_hash.0), code.clone());
-                            }
-                            Entry::Occupied(mut entry) => {
-                                entry.insert(code.clone());
-                            }
-                        }
+                    if !code.is_empty() && !db.contracts.contains_key(&account.info.code_hash) {
```

Comment:
It is more efficent to have `Entry`, this is like double hashing code_hash as it needs two accesses to db.